-------------

# My Objective is to find the relationship between `Age` and `Avg Glucose Level` in the dataset of Stroke Prediction

-----------


### The dataset is imported from the link below:
<img src="https://media.giphy.com/media/HEp2Dm0CBT72LRuFIw/giphy.gif" style="width:20%;height:30%;float:left "/> 




  # [Link of Dataset of Wine Quality Test](https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset)

![](stroke.jpg)

## Snapshot of Dataset
![](stroke.png)

## Importing numpy, pandas, mathplot and other methods

In [1]:
import pandas as pd #pandas is used for data manipulation
import numpy as np #mathematical calculation like matrix vector and all
import matplotlib.pyplot as plt #data visualisation graph
%matplotlib inline
import seaborn as sns  #data visualisation in graph
import plotly.express as px   #web-basedd visualisation in graph

Firstly, Reading the file

In [2]:
df = pd.read_csv("healthcare-dataset-stroke-data.csv")

In [3]:
df.head(6)   #It shows the data from the first five rows as default but user can change it as how they want

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   
5  56669    Male  81.0             0              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   
5        Private          Urban             186.21  29.0  formerly smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1  
5       1

In the above table,
- The attributes of data is shown with its data from the first 6 rows

In [4]:
df.tail(7)  #It shows the data from last

id  gender   age  hypertension  heart_disease ever_married  \
5103  22127  Female  18.0             0              0           No   
5104  14180  Female  13.0             0              0           No   
5105  18234  Female  80.0             1              0          Yes   
5106  44873  Female  81.0             0              0          Yes   
5107  19723  Female  35.0             0              0          Yes   
5108  37544    Male  51.0             0              0          Yes   
5109  44679  Female  44.0             0              0          Yes   

          work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
5103        Private          Urban              82.85  46.9          Unknown   
5104       children          Rural             103.08  18.6          Unknown   
5105        Private          Urban              83.75   NaN     never smoked   
5106  Self-employed          Urban             125.20  40.0     never smoked   
5107  Self-employed          Rural              82.99  30.6     never smoked   
5108        Private          Rural             166.29  25.6  formerly smoked   
5109       Govt_job          Urban              85.28  26.2          Unknown   

      stroke  
5103       0  
5104       0  
5105       0  
5106       0  
5107       0  
5108       0  
5109       0

In the above table,
- The attributes of data is shown from the last 7 rows

---------

## Summary Statistics

---------

In [5]:
df.describe()  #It distributes the data in the dataset

id          age  hypertension  heart_disease  \
count   5110.000000  5110.000000   5110.000000    5110.000000   
mean   36517.829354    43.226614      0.097456       0.054012   
std    21161.721625    22.612647      0.296607       0.226063   
min       67.000000     0.080000      0.000000       0.000000   
25%    17741.250000    25.000000      0.000000       0.000000   
50%    36932.000000    45.000000      0.000000       0.000000   
75%    54682.000000    61.000000      0.000000       0.000000   
max    72940.000000    82.000000      1.000000       1.000000   

       avg_glucose_level          bmi       stroke  
count        5110.000000  4909.000000  5110.000000  
mean          106.147677    28.893237     0.048728  
std            45.283560     7.854067     0.215320  
min            55.120000    10.300000     0.000000  
25%            77.245000    23.500000     0.000000  
50%            91.885000    28.100000     0.000000  
75%           114.090000    33.100000     0.000000  
max           271.740000    97.600000     1.000000

In the above dataset, <br>
For Age,
- average is 43.23
- standard deviation is 22.61
- min is 0.08
- max is 82 <br>

whereas <br>
For avg glucose level,
- average is 106.15
- standard deviation is 45.28
- min is 55.12
- max is 271.74

---------------

## Complete information about the dataset 

------------

In [6]:
df.info() #it shows all the information about the dataset

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


In this dataset,
- Class is pandas.core.frame.Dataframe
- has 12 features
- range index of 5110 entries, 0 to 3109
- has 3 float, 4 integers, and 5 object
- memory usage of 479.2+ KB

-------

## INspecting missing values in dataset

------------

In [8]:
print(df.isnull())

         id  gender    age  hypertension  heart_disease  ever_married  \
0     False   False  False         False          False         False   
1     False   False  False         False          False         False   
2     False   False  False         False          False         False   
3     False   False  False         False          False         False   
4     False   False  False         False          False         False   
...     ...     ...    ...           ...            ...           ...   
5105  False   False  False         False          False         False   
5106  False   False  False         False          False         False   
5107  False   False  False         False          False         False   
5108  False   False  False         False          False         False   
5109  False   False  False         False          False         False   

      work_type  Residence_type  avg_glucose_level    bmi  smoking_status  \
0         False           False              F

In [9]:
#Inspect missing values in the dataset
print(df.isnull().values.sum())

# Replace the ' 's with NaN
df = df.replace(" ",np.NaN)

#Impute the missing values with mean imputation
df = df.fillna(df.mean())

# Count the number of NaNs in the dataset to verify
print(df.isnull().values.sum())

201
0


C:\Users\Pragyan\AppData\Local\Temp\ipykernel_1500\3168217728.py:8: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  df = df.fillna(df.mean())
